File: repl/java/phone-mapping-for-letters/map_letters_to_phone_number.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class map_letters_to_phone_number {

    public static void main(String[] args) {
        start_phone_converter();
    }

    public static String convert_letters_to_phone_digits(String input_string) {
        // Standard telephone keypad mapping
        Map<Character, Character> lookup_table = new HashMap<>();
        setup_lookup_table(lookup_table, "abc", '2');
        setup_lookup_table(lookup_table, "def", '3');
        setup_lookup_table(lookup_table, "ghi", '4');
        setup_lookup_table(lookup_table, "jkl", '5');
        setup_lookup_table(lookup_table, "mno", '6');
        setup_lookup_table(lookup_table, "pqrs", '7');
        setup_lookup_table(lookup_table, "tuv", '8');
        setup_lookup_table(lookup_table, "wxyz", '9');

        StringBuilder result = new StringBuilder();

        for (char c : input_string.toLowerCase().toCharArray()) {
            if (lookup_table.containsKey(c)) {
                result.append(lookup_table.get(c));
            } else if (Character.isDigit(c)) {
                // Keep existing numbers as they are
                result.append(c);
            }
            // Skip spaces, punctuation, or symbols as per original logic
        }

        return result.toString();
    }

    private static void setup_lookup_table(Map<Character, Character> table, String chars, char digit) {
        for (char c : chars.toCharArray()) {
            table.put(c, digit);
        }
    }

    public static void start_phone_converter() {
        Scanner input_scanner = new Scanner(System.in);
        System.out.println("--- Phone Scheme Converter (Java) ---");
        System.out.println("Type 'exit' to quit.");

        while (true) {
            System.out.print("\nEnter text to convert: ");
            String user_input = input_scanner.nextLine();

            if (user_input.equalsIgnoreCase("exit")) {
                System.out.println("Exiting...");
                break;
            }

            String converted_number = convert_letters_to_phone_digits(user_input);
            System.out.println("Phone Number: " + converted_number);
        }
        
        input_scanner.close();
    }
}

File: repl/python/phone-mapping-for-letters/map-letters-to-phone-number.py
def convert_letters_to_phone_digits(input_string):
    # Mapping based on the international standard telephone keypad
    phone_map = {
        'abc': '2', 'def': '3', 'ghi': '4',
        'jkl': '5', 'mno': '6', 'pqrs': '7',
        'tuv': '8', 'wxyz': '9'
    }
    
    # Create a fast lookup dictionary (char -> digit)
    lookup_table = {char: digit for chars, digit in phone_map.items() for char in chars}
    
    result = []
    
    for char in input_string.lower():
        if char in lookup_table:
            result.append(lookup_table[char])
        elif char.isdigit():
            # Keep existing numbers as they are
            result.append(char)
        else:
            # Skip spaces, punctuation, or symbols
            continue
            
    return "".join(result)

def start_phone_converter():
    print("--- Phone Scheme Converter ---")
    print("Type 'exit' to quit.")
    
    while True:
        user_input = input("\nEnter text to convert: ")
        
        if user_input.lower() == "exit":
            print("Exiting...")
            break
            
        converted_number = convert_letters_to_phone_digits(user_input)
        print(f"Phone Number: {converted_number}")

if __name__ == "__main__":
    start_phone_converter()